xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -13171,8 +13171,9 @@ glusterd_update_mntopts(char *brick_path, glusterd_brickinfo_t *brickinfo)
 }
 
 int
-glusterd_get_value_for_vme_entry(struct volopt_map_entry *vme, char **def_val)
+glusterd_get_value_for_vme_entry(void *vmep, char **def_val)
```

Comment:
OK I was thinking why are you doing this. My guess is that you need this function in broader scope and for that purpose you add a delcaration in a header, but the type of the argument, `struct volopt_map_entry` is not available in the  header.

But this does not seem to be correct: all callers do know about `struct volopt_map_entry` and the callers pass such a pointer to the function. So it's not an opaque pointer type for which `void *` is suitable.

Why don't you just put the declaration of the function to the same place where the definition of the struct is given, ie. _glusterd-volgen.h_?

Reply:
I can't recall correctly.....but AFAIR it was because of the way in which files are imported across the "glusterd" code, it starts giving redefinition errors. Hence had to resort to this method.